js/utils/helpers.js

```diff
@@ -2760,6 +2770,56 @@
 
             if (catalogSnapshot.length === 2) obj.$catalog.version = catalogSnapshot[1];
         }
+
+        if (module._session) {
+            var session = module._session;
+            obj.$session = {
+                display_name: session.client.display_name,
+                email: session.client.email,
+                full_name: session.client.full_name,
+                id: session.client.id,
+                identities: session.client.identities,
+                attributes: []
+            };
+
+            session.attributes.forEach(function (attr) {
+                var tempAttr = attr;
+                // NOTE: the current assumption is that everything in session.attributes is a globus group or an identity
+                // a globus group if:
+                //   - display_name is defined
+                //   - display_name is not the same as the user's display_name
+                //   - current id is not in the identities array
+                if (attr.display_name && attr.display_name !== session.client.display_name && session.client.identities.indexOf(attr.id) == -1) {
+                    if (attr.id.indexOf("https://auth.globus.org/") === 0) {
+                        // assume id is always "https://auth.globus.org/ff766864-a03f-11e5-b097-22000aef184d"
+                        tempAttr.webpage = "https://app.globus.org/groups/" + attr.id.substring(24) + "/about";
+                        tempAttr.type = "globus_group";
+                    } else {
+                        // NOTE: currently no example for this case
+                        // tempAttr.type = "other_group";
+                    }
+                } else {
+                    tempAttr.type = "identity";
+                }
+
+                var matchIdx = null;
+                // determine if 'tempAttr' exists in $session.attributes
+                matchIdx = obj.$session.attributes.findIndex(function (targetAttr) {
+                    return targetAttr.id === tempAttr.id;
+                });
+
+                if (matchIdx !== -1) {
```

Comment:
I'm not entirely sure what you're doing here. Are you imposing a rule that the "id" attribute must be unique in the `session.attributes`?  Wasn't the goal just to add some extra attributes (type, webpage, etc) to each one?

Reply:
We discussed merging identities that were the same. I documented that in the issue when we had the discussion about it.

Reply:
I forgot that's what we agreed on. In that case I would say add some small comments just explaining what this code is supposed to do